### ISSUE TITLE
Gradio Audio Component Issue for Longer Audio

### ISSUE THREAD

**Issue by SamridhiVaid** [gradio-app/gradio#8602]
### Describe the bug

I've encountered an issue with the Gradio Audio component and haven't received any response after posting about it. I'm using Gradio version 4.31.5 and have tried replicating the issue on other versions as well, but couldn't find any solutions online. 

I'm using the Gradio UI to record audio, which is then input into WhisperX, and the transcript generated is processed by an LLM to produce a note. This setup is hosted on a server. The problem arises when I record a longer audio greater than 4 minutes: I have to wait 40-50 seconds, depending on the length of the audio, before I can hit the submit button to transcribe it. The logs show multiple POST requests like:

```
INFO:     172.21.88.12:0 - "POST /gradio/upload HTTP/1.1" 200 OK
INFO:     172.21.124.64:0 - "POST /gradio/upload HTTP/1.1" 200 OK
INFO:     172.21.88.128:0 - "POST /gradio/upload HTTP/1.1" 200 OK
INFO:     172.21.122.64:0 - "POST /gradio/upload HTTP/1.1" 200 OK
INFO:     172.21.88.128:0 - "POST /gradio/upload HTTP/1.1" 200 OK
INFO:     172.21.104.64:0 - "POST /gradio/upload HTTP/1.1" 200 OK
INFO:     172.21.87.192:0 - "POST /gradio/queue/join HTTP/1.1" 200 OK
INFO:     172.21.104.64:0 - "POST /gradio/upload HTTP/1.1" 200 OK
```
But before these post request start, I have to wait. And then only after a minute or so, does the waveform for the recorded audio file appears. If I submit my audio file before that I get error since my code cannot find audio file.

Is there a way to speed up the upload to the Gradio endpoint? Alternatively, can I show a loading symbol so that users know to wait before hitting the submit button? Why do these uploads take so long? I've also replicated this issue on other online Gradio audio applications, which error out if I hit submit immediately. 

### Have you searched existing issues?  🔎

- [X] I have searched and found no existing issues

### Reproduction

```python
import gradio as gr

```
Use any existing radio application. Do a recording of 4-5 mins and click submit button. https://www.gradio.app/guides/real-time-speech-recognition

### Screenshot

<img width="1728" alt="Screenshot 2024-06-24 at 7 32 39 PM" src="https://github.com/gradio-app/gradio/assets/29611466/7843675a-f09f-4550-a51e-665e6176192f">

### System Info

```shell
Gradio Version: 4.31.5
```

### Severity

Blocking usage of gradio

**Comment by ROBERT-MCDOWELL** [comment]
is anyone working on it?

**Comment by JackismyShephard** [comment]
@abidlabs  @pngwn I have a partially related issue: loading audio into an audio component takes a long time when you have many audio components in a gradio app. I suspect this might have something to do with rendering the wave form in the audio component. It is quite a distraction from a UI perspective as audio can take upwards of 5 seconds to load.

**Comment by freddyaboulton** [comment]
Hi @SamridhiVaid @ROBERT-MCDOWELL, we already display upload progress in the UI. Though there currently is a bug on spaces where it does not show up always. I opened #11102 for this.

@JackismyShephard can you open a new issue with a minimal reproducer? I think it's related to wavesurfer having to load the entire file into a blob before rendering the waveform. Not sure if there is a way around that.

**Comment by JackismyShephard** [comment]
@freddyaboulton That is interesting. In the worst case it might be nice to then just have the possibility to disable rendering the waveform. I will open a new issue

**Comment by abidlabs** [comment]
@JackismyShephard you can disable rendering the waveform by setting `show_waveform=False` ([link](https://www.gradio.app/docs/gradio/audio#param-waveformoptions-show-recording-waveform)) in the `waveform_options` parameter [link](https://www.gradio.app/docs/gradio/audio#param-audio-waveform-options) of `gr.Audio`

**Comment by freddyaboulton** [comment]
See @abidlabs 's comment here https://github.com/gradio-app/gradio/issues/8602#issuecomment-2840752285 @ROBERT-MCDOWELL .

**Comment by ROBERT-MCDOWELL** [comment]
@freddyaboulton 
it's show_recording_wavform, show_wavform does not exist.
